prow/github/client.go

```diff
@@ -2122,6 +2124,45 @@ func (c *client) GetPullRequest(org, repo string, number int) (*PullRequest, err
 	return &pr, err
 }
 
+func (c *client) GetFailedActionRunsByHeadBranch(org, repo, branchName, headSHA string) ([]WorkflowRun, error) {
+	durationLogger := c.log("GetJobsByHeadBranch", org, repo)
+	defer durationLogger()
+
+	var runs WorkflowRuns
+
+	_, err := c.request(&request{
+		accept:    "application/vnd.github.v3+json",
+		method:    http.MethodGet,
+		path:      fmt.Sprintf("/repos/%s/%s/actions/runs?status=failure&event=pull_request&branch=%s", org, repo, url.QueryEscape(branchName)),
```

Comment:
nit: can we perhaps create a URL object, encode query parameters directly, then stringify it? this would help to make the construction here easier to follow and ensure that you don't have to remember to query-escape the data